provisioner/src/lib.rs

```diff
@@ -457,6 +555,301 @@ impl Provisioner for MyProvisioner {
+
+        self.delete_saved_access_key().await?;
+
+        Ok(())
+    }
+
+    async fn delete_iam_identity(&self) -> Result<DeleteUserOutput, Error> {
+        let user = self
+            .iam_client
+            .delete_user()
+            .user_name(self.get_iam_identity_user_name().await)
+            .send()
+            .await
+            .map_err(Error::DeleteIAMUser)?;
+        Ok(user)
+    }
+
+    async fn attach_user_policy(&self) -> Result<(), Error> {
+        self.iam_client
+            .attach_user_policy()
+            .user_name(self.get_iam_identity_user_name().await)
+            .policy_arn(self.get_policy_arn().await?)
+            .send()
+            .await
+            .map_err(Error::AttachUserPolicy)?;
+        Ok(())
+    }
+
+    async fn detach_user_policy(&self) -> Result<(), Error> {
+        self.iam_client
+            .detach_user_policy()
+            .user_name(self.get_iam_identity_user_name().await)
+            .policy_arn(self.get_policy_arn().await?)
+            .send()
+            .await
+            .map_err(Error::DetachUserPolicy)?;
+        Ok(())
+    }
+
+    async fn get_iam_identity_user_name(&self) -> String {
+        // max characters is 64
+        format!("{}-dynamo-user", self.prefix)
+    }
+
+    async fn create_iam_identity(&self) -> Result<(), Error> {
+        match self
+            .iam_client
+            .create_user()
+            .user_name(self.get_iam_identity_user_name().await)
+            .send()
+            .await
+        {
+            Ok(_) => {}
+            Err(e) => match e.into_service_error() {
+                CreateUserError::EntityAlreadyExistsException(_) => {}
+                e => {
+                    return Err(Error::CreateIAMUser(e));
+                }
+            },
+        };
+        Ok(())
+    }
+
+    async fn get_saved_access_key(&self) -> Option<(String, String)> {
+        if let Ok(file) = std::fs::File::open(self.get_access_key_file_name()) {
+            let mut lines = std::io::BufReader::new(file).lines();
+
+            if let Some(Ok(access_key_id)) = lines.next() {
+                if let Some(Ok(secret_access_key)) = lines.next() {
+                    return Some((access_key_id, secret_access_key));
+                }
+            }
+        }
+
+        None
+    }
+
+    fn get_access_key_file_name(&self) -> String {
+        format!("{}.txt", self.prefix)
+    }
+
+    async fn delete_saved_access_key(&self) -> Result<(), std::io::Error> {
+        std::fs::remove_file(self.get_access_key_file_name())?;
+        Ok(())
+    }
+
+    async fn save_access_key(
+        &self,
+        access_key_id: &str,
+        secret_access_key: &str,
+    ) -> Result<(), std::io::Error> {
+        use std::io::prelude::*;
+        let mut file = File::create(self.get_access_key_file_name())?;
+        let contents = format!("{}\n{}", access_key_id, secret_access_key);
+        file.write_all(contents.as_bytes())?;
```

Comment:
@mikegin , @chesedo This is to enhance my knowledge but not comment on a PR. Sorry for spamming here.
As I am not well aware of shuttle deployment architecture, I am having a little hard time understanding how writing data to a file in a container will provide persistence. What if the provisioner container gets redeployed due to some reason? or are we okay to lose this data and the file is just a good to have cache?

Reply:
The provisioner will have its own docker volume, which will persist across invocations. This wasn't the case before but will be introduced through this PR.